git_machete/cmd.py

```diff
@@ -252,8 +260,8 @@ def popen_cmd(cmd, *args, **kwargs):
 # Git core
 
 
-def cmd_shell_repr(cmd, *args, **kwargs):
-    def shell_escape(arg):
+def cmd_shell_repr(cmd: str, *args: str, **kwargs: Dict[str, str]) -> str:
```

Comment:
Actually... why `kwargs` are `Dict[str, str]` but `args` are `str` and not `List[Str]`? 🤔

Reply:
That seems inconsistent...

Reply:
Results of `mypy` while annotating `kwargs` as `str`:
```
➜  git-machete git:(refactor/mypy) ✗ mypy --config-file mypy.ini git_machete
git_machete/cmd.py:274: error: Need type annotation for 'env'
git_machete/cmd.py:276: error: Item "str" of "Union[str, Dict[Any, Any]]" has no attribute "items"
git_machete/cmd.py:283: error: Argument 4 to "cmd_shell_repr" has incompatible type "**Dict[str, Dict[str, str]]"; expected "str"
git_machete/cmd.py:290: error: Argument 4 to "cmd_shell_repr" has incompatible type "**Dict[str, Dict[str, str]]"; expected "str"
```

Results of `mypy` while annotating `args` as `List[str]`:
```
➜  git-machete git:(refactor/mypy) ✗ mypy --config-file mypy.ini git_machete
git_machete/cmd.py:220: error: Argument 2 to "cmd_shell_repr" has incompatible type "*Tuple[str, ...]"; expected "List[str]"
git_machete/cmd.py:241: error: Argument 2 to "cmd_shell_repr" has incompatible type "*Tuple[str, ...]"; expected "List[str]"
git_machete/cmd.py:277: error: Argument 1 to "map" has incompatible type "Callable[[str], str]"; expected "Callable[[List[str]], str]"
git_machete/cmd.py:283: error: Argument 2 to "cmd_shell_repr" has incompatible type "str"; expected "List[str]"
git_machete/cmd.py:283: error: Argument 3 to "cmd_shell_repr" has incompatible type "*Tuple[str, ...]"; expected "List[str]"
git_machete/cmd.py:290: error: Argument 2 to "cmd_shell_repr" has incompatible type "str"; expected "List[str]"
git_machete/cmd.py:290: error: Argument 3 to "cmd_shell_repr" has incompatible type "*Tuple[str, ...]"; expected "List[str]"
```
---


Specifying the type of `*args` tells us what EACH "arg" is.
If we use `str` then each of them is a string.
If we use `List[str]` then they are all to be a list of strings.

The same for `kwargs`.

In our case, the "kwarg" is not a type of `str`. It is `Dict[str, str]`.
Why? See how we use it (the type of `kwargs.get`)...
```
env: Dict[str, str] = kwargs.get("env", {})
    # ...
    env_repr = [k + "=" + shell_escape(v) for k, v in env.items() if k not in os.environ]
```


--- 

Refs: 
https://www.python.org/dev/peps/pep-0484/#arbitrary-argument-lists-and-default-argument-values
https://stackoverflow.com/questions/37031928/type-annotations-for-args-and-kwargs

Reply:
Aww crap I forgot about this `env` thing 🤦🏻

Reply:
There might be different `str`-typed params in the future (or maybe they were in the past?) but yeah, `Dict[str, str]` FTW as for now